projects/kit/components/input-time/input-time.directive.ts

```diff
@@ -0,0 +1,165 @@
+    TuiWithTextfield,
+} from '@taiga-ui/core/components/textfield';
+import {
+    TuiDropdownDirective,
+    tuiDropdownEnabled,
+    tuiDropdownOpen,
+} from '@taiga-ui/core/directives/dropdown';
+import {tuiAsAuxiliary} from '@taiga-ui/core/tokens/auxiliary';
+import {TuiSelectOption} from '@taiga-ui/kit/components/select';
+import {TUI_TIME_TEXTS} from '@taiga-ui/kit/tokens';
+import {tuiMaskito} from '@taiga-ui/kit/utils';
+
+import {TUI_INPUT_TIME_OPTIONS} from './input-time.options';
+
+@Directive({
+    standalone: true,
+    selector: 'input[tuiInputTime]',
+    providers: [
+        tuiAsControl(TuiInputTimeDirective),
+        tuiAsTextfieldAccessor(TuiInputTimeDirective),
+        tuiAsAuxiliary(TuiInputTimeDirective),
+        tuiValueTransformerFrom(TUI_INPUT_TIME_OPTIONS),
+        tuiAsOptionContent(TuiSelectOption),
+    ],
+    hostDirectives: [TuiWithTextfield, MaskitoDirective],
+    host: {
+        '[disabled]': 'disabled()',
+        '[attr.inputMode]': 'inputMode()',
+        '(blur)': 'onTouched()',
+        '(click)': 'toggle()',
+    },
+})
+export class TuiInputTimeDirective
+    extends TuiControl<TuiTime | null>
+    implements TuiTextfieldAccessor<TuiTime | null>
+{
+    private readonly textfield = inject(TuiTextfieldDirective);
+    private readonly dropdown = inject(TuiDropdownDirective);
+    private readonly open = tuiDropdownOpen();
+    private readonly options = inject(TUI_INPUT_TIME_OPTIONS);
+    private readonly fillers = toSignal(inject(TUI_TIME_TEXTS));
+    private readonly acceptableValues = signal<readonly TuiTime[]>([]);
+
+    protected readonly icon = tuiTextfieldIconBinding(this.options);
+    protected readonly dropdownEnabled = tuiDropdownEnabled(
+        computed(() => !this.nativePickerEnabled && this.interactive()),
+    );
+
+    protected readonly filler = tuiDirectiveBinding(
+        TuiTextfieldComponent,
+        'fillerSetter',
+        computed((fillers = this.fillers()) => fillers?.[this.timeMode()] ?? ''),
+        {},
+    );
+
+    protected readonly inputMode = computed(() =>
+        this.timeMode().includes('AA') ? 'text' : 'numeric',
+    );
+
+    protected readonly mask = tuiMaskito(
+        computed(() =>
+            this.computeMask({
+                ...this.options,
+                mode: this.timeMode(),
+                step: this.interactive() && !this.dropdown._content() ? 1 : 0,
+            }),
+        ),
+    );
+
+    protected readonly valueEffect = effect(() => {
+        const value = this.textfield.value();
+        const time =
+            value.length === this.timeMode().length ? TuiTime.fromString(value) : null;
+        const newValue =
+            this.acceptableValues().length && time
+                ? this.findNearestTime(time, this.acceptableValues())
+                : time;
+
+        this.control?.control?.updateValueAndValidity({emitEvent: false});
+        this.onChange(newValue);
+
+        if (newValue && newValue !== time) {
```

Comment:
Wouldn't this cause an infinite loop due to time being recreated from the string value an newValue being picked from the list, so === will never be true?